src/components/Scrollable/Scrollable.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import styled from 'styled-components';
+
+const StyledDiv = styled.div`
+  height: 100%;
+  color: ${props => props.theme.color};
+
+  & > div {
+    height: 100%;
+    overflow: scroll;
+  }
+
+  &::before,
+  &::after {
+    font-family: 'Font Awesome 5 Free';
+    font-size: 0.8rem;
+    position: absolute;
+    left: ${props => (props.position === 'left' ? '3px' : 'unset')};
+    right: ${props => (props.position === 'right' ? '3px' : 'unset')};
+    z-index: 1000;
+
+    font-weight: 900;
+  }
+
+  &.up::before {
+    content: '\f106';
+    top: 0;
+  }
+
+  &.down::after {
+    content: '\f107';
+    bottom: 0;
+  }
+`;
+
+class Scrollable extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      displayUp: false,
+      displayDown: false
+    };
+
+    this.checkScroll = this.checkScroll.bind(this);
+
+    this.scrollable = React.createRef();
+  }
+
+  componentDidMount() {
+    this.checkScroll();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.children.length !== prevProps.children.length)
+      this.checkScroll();
+  }
+
+  checkScroll() {
+    const element = this.scrollable.current;
+
+    if (element && window.getComputedStyle(element).overflowY !== 'hidden') {
```

Comment:
As you use `window`, the `checkScroll` method becomes unpure and not testable. You should use `window` in an helper, which can be mocked during the tests.

Reply:
Hmm... I'm not sure to understand what you propose :/

What I meant is:
* The `checkScroll` method calls `window` (which is an external dependency as it not provided by an argument)
* Plus, `window` is related to the browser environment so it is not available in tests without `jsdom` for instance

So I proposed to put that call in an external function/helper which can be mocked during the tests.

Eg:

```js
// src/helpers/browser.js
export function isOverflowYHidden(element) {
  return window.getComputedStyle(element).overflowY !== 'hidden';
}
```

And during your tests:

```js
jest.mock('helpers/browser.js', () => {
  isOverflowYHidden: () => true,
})
```

Or `false`, whatever. But the further tests will internally call `isOverflowYHidden` and it will always return the value you set in the mock. The behavior is controlled, the earth is safe 💪